src/pages/AddEditTask.tsx

```diff
@@ -40,6 +42,41 @@ const AddTask = (props: AddTaskProps) => {
         ];
     });
 
+    // if the task already exists, populate the fields
+    useEffect(() => {
+        console.log(params)
+        async function fillBoxes(task_id: number) {
+            const task = await getTask(task_id);
+            setTitle(task.title);
+            setDueDate(task.due_date);
+            setDescription(task.description);
+            setDone(task.completed);
+            setColors(colors.map((color) => {
+                if (color.color === task.color) {
+                    return { color: color.color, selected: true };
+                }
+                return { color: color.color, selected: false };
+            }));
+            
+            const room = await getRoom(task.room_id);
+            console.log(room)
+            setSelectedRoom(room);
+
+            const tags_of_task = await getTagsOfTask(task_id);
+            const tags = await Promise.all(tags_of_task.map(async (tag_of_task) => {
+                return await getTag(tag_of_task.tag_name);
+            }));
```

Comment:
Nice. It's really satisfying to see all of these roll in and appear on the screen.

If you want, you can use `Promise.all` to asynchronously run these queries in parallel. Your call. See [here for an example](https://github.com/MrEminent42/casamico/blob/1af1f7b617d5288f3a5e1e5d2dcefb5f052bdab0/src/pages/Tasks.tsx#L49).

Reply:
Actually I guess you need the first call to finish before you can do the second, since you need the room_id. So ignore this if it's not relevant.